libs/vkd3d/command.c

```diff
@@ -4069,6 +4069,42 @@ static bool d3d12_command_list_update_compute_pipeline(struct d3d12_command_list
     return true;
 }
 
+static bool d3d12_command_list_update_raygen_pipeline(struct d3d12_command_list *list)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = &list->device->vk_procs;
+
+    if (list->current_pipeline != VK_NULL_HANDLE)
+        return true;
+
+    if (!list->rt_state)
+    {
+        WARN("Pipeline state %p is not a raygen pipeline.\n", list->rt_state);
+        return false;
+    }
+
+    if (list->command_buffer_pipeline != list->rt_state->pipeline)
+    {
+        VK_CALL(vkCmdBindPipeline(list->vk_command_buffer,
+                VK_PIPELINE_BIND_POINT_RAY_TRACING_KHR,
+                list->rt_state->pipeline));
+        list->command_buffer_pipeline = list->rt_state->pipeline;
+        /* Ensures we update the pipeline stack size.
+         * It is somewhat overkill to introduce this to the dynamic state bitmask system. */
+        list->dynamic_state.pipeline_stack_size = UINT32_MAX;
```

Comment:
I am not a fan of this, why not just move the `vkCmdSetRayTracingPipelineStackSizeKHR` into this block and scrap the `dynamic_state` tracking of `pipeline_stack_size`entirely?

D3D12 only has this in PSO state so, idk why it's being tracked in dynamic here when we can resolve this at the same time we bind the pipeline.

Reply:
The stack size can change on the fly after binding the PSO, so we have to call SetPipelineStackSize every time. I guess it can't hurt all that much.

Reply:
Oh it can? I wasn't aware of that. Makes sense then.

Reply:
Fixed.